File: app/src/main/java/com/example/readative/room/entity/Series.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(indices = {@Index(value = {"series_name"}, unique = true)})
public class Series {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "series_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "series_name")
    public String name;
}

File: app/src/main/java/com/example/readative/feature_book/data/data_source/models/Series.kt
package com.example.readative.feature_book.data.data_source.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "series",
    indices = [
        Index(value = ["name"], unique = true)
    ]
)
data class Series(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "name") val name: String
) : ReadativeEntity